syft/pkg/cataloger/kernel/parse_linux_kernel_module_file.go

```diff
@@ -0,0 +1,182 @@
+package kernel
+
+import (
+	"debug/elf"
+	"fmt"
+	"strings"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/internal/unionreader"
+	"github.com/anchore/syft/syft/source"
+)
+
+func findLinuxKernelModules(resolver source.FileResolver, pkgLocations *source.LocationSet, kernelVersion string) ([]pkg.LinuxKernelModuleMetadata, error) {
+	locations, err := resolver.FilesByGlob(fmt.Sprintf("**/lib/modules/%s/**/*.ko", kernelVersion))
```

Comment:
I've noticed that there are some kernel modules that are distributed as `.ko.xz` files. Should we attempt to catalog these? is this common of distributions? Or is `.ko` more common?

Reply:
`.ko` is more common, as far as I know, but compressed modules are valid as well.

I think that this works as is (subject to the kernel module packages, which I am awaiting your push to test it), so let's use it, and not delay it for `.ko.xz` files.  We always can have a separate small PR to add it, which will be easy to do .